### PR TITLE
[path_provider] De-flake getExternalStorageDirectories test

### PR DESCRIPTION
**Pull request by gmackall**
Fixes the flaky path provider test. 

Directories.listSync will sometimes not return the files in the directory (as discussed in https://github.com/flutter/flutter/issues/139378), but the files are there and (after letting it loop for quite a while and comparing to current flake) they always appear in the output of `Process.runSync('ls', ...)`. 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [ ] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by stuartmorgan** [comment]
> Directories.listSync will sometimes not return the files in the directory (as discussed in [flutter/flutter#139378](https://github.com/flutter/flutter/issues/139378)), but the files are there and (after letting it loop for quite a while and comparing to current flake) they always appear in the output of `Process.runSync('ls', ...)`.

That's super weird. Is there a Dart issue filed that we can reference in a comment?

File: packages/path_provider/path_provider/example/integration_test/path_provider_test.dart
Comment (stuartmorgan) [inline packages/path_provider/path_provider/example/integration_test/path_provider_test.dart:120]: For the app-facing package we should probably make this codepath Android-only, and have other platforms keep using `listSync`.

Comment (gmackall) [reply packages/path_provider/path_provider/example/integration_test/path_provider_test.dart]: Changed!

**Review by stuartmorgan — APPROVED** [review]
LGTM once autoformat is happy.

**Comment by gmackall** [comment]
> LGTM once autoformat is happy.

I don't know why I assumed that was the changelog/version checker complaining 🤦. Thanks for taking another look, I'll fix that and then land

**Comment by gmackall** [comment]
Tree is blocked on two failures, one of which is this test failing (https://github.com/flutter/packages/pull/5619 only disabled the test in `path_provider_android` but this same test is also run by the app-facing package on android).

Should this land on red to fix, or does the other blocker need to be resolved first?

**Pull request merged**
